src/config.h

```diff
@@ -188,6 +189,14 @@ extern RSConfigOptions RSGlobalConfigOptions;
  */
 void RSConfigOptions_AddConfigs(RSConfigOptions *src, RSConfigOptions *dst);
 
+/**
+ * Register a new external trigger for configuration changes
+ * @param trigger the trigger function
+ * @param configs an array of configuration names that trigger the function.
+ *                The array must be NULL-terminated.
+ */
```

Comment:
I don't see the point of explaining `RSConfigExternalTrigger` here. It only get a non-const reference because we call `getOrCreateRealConfig` and might have to link the cluster global configs. Nobody stops us from editing the configuration struct but that is not the intension here (at least for now)